src/components/Item.jsx

```diff
@@ -20,29 +21,22 @@ const Item = ({
         });
     }
   };
+  const calculateLatestInterval = () => {
+    let currentDate = new Date().getTime();
 
-  const checkDate = (lastPurchasedDate) => {
-    if (lastPurchasedDate === null) {
-      return false;
-    } else {
-      const day = 60 * 60 * 24;
-      const now = Date.now() / 1000;
-      return now - lastPurchasedDate.seconds < day;
-    }
+    let interval = Math.round(
+      // convert from date to seconds
+      (currentDate / 1000 - lastPurchasedDate.seconds) /
+        // convert from seconds to day
+        (60 * 60 * 24),
+    );
+    return interval;
```

Comment:
imo, I'd remove rounding at all.

Reply:
We're going  to remove the rounding here because the `calculateEstimate` function will already round the result. If we round it up early with `Math.ceil`, there will eventually be 1 extra day right?